shared/src/androidMain/kotlin/org/hyperskill/app/purchases/domain/AndroidPurchaseManager.kt

```diff
@@ -45,6 +48,18 @@ class AndroidPurchaseManager(
             Purchases.sharedInstance.awaitLogIn(userId.toString())
         }
 
+    @Suppress("TooGenericExceptionCaught")
+    override suspend fun canMakePayments(): Boolean =
+        suspendCoroutine { continuation ->
+            try {
+                Purchases.canMakePayments(application) { result ->
+                    continuation.resume(result)
+                }
+            } catch (e: Exception) {
+                continuation.resumeWithException(e)
```

Comment:
If this function will be called without try-catch app will crash?

For example `AppActionDispatcher`:
```kotlin
is InternalAction.FetchPaymentAbility ->
    onNewMessage(InternalMessage.PaymentAbilityResult(purchaseInteractor.canMakePayments()))
```

Reply:
Yeah, you are right. It's better to return Result instead of a raw value.

Reply:
> Yeah, you are right. It's better to return Result instead of a raw value.

or you can return `false` when error catched, we are not handling/logging error upstairs